contracts/contracts/StakingEscrow.sol

```diff
@@ -202,7 +158,7 @@ contract StakingEscrow is Upgradeable, IERC900History {
         IVendingMachine _vendingMachine
     ) {
         require(_token.totalSupply() > 0 &&
-            _tStaking.stakedNu(address(0)) == 0 &&
+            _tStaking.getApplicationsLength() != 0 &&
```

Comment:
why not checking that `_tStaking.token() == address(_token)` instead? seems more natural at first

Reply:
I wanted, but it's not part of `IStaking`